app/jobs/ingest_job.rb

```diff
@@ -48,12 +48,16 @@ def perform(model_params:, background_job_result:, signed_ids: {}, globus_ids: {
     # Create workflow destroys existing steps if called again, so need to check if already created.
     Workflow.create_unless_exists(druid, 'registrationWF', version: 1, priority: priority)
 
-    StageBlobs.stage(signed_ids, druid) do
-      Workflow.create_unless_exists(druid, 'accessionWF', version: 1, priority: priority) if start_workflow
-    end
-
-    StageGlobus.stage(globus_ids, druid) do
-      Workflow.create_unless_exists(druid, 'accessionWF', version: 1, priority: priority) if start_workflow
+    if signed_ids.any?
+      StageBlobs.stage(signed_ids, druid) do
+        Workflow.create_unless_exists(druid, 'accessionWF', version: 1, priority: priority) if start_workflow
+      end
+    elsif globus_ids.any?
```

Comment:
It should be possible to have both signed_ids and globus_ids.

Reply:
I think you probably want something like:
```
StageBlobs.stage(signed_ids, druid)
StageGlobus.stage(globus_ids, druid)
Workflow.create_unless_exists(druid, 'accessionWF', version: 1, priority: priority) if start_workflow && (signed_ids.any? || globus_ids.any?)
```

Reply:
I can't really figure out the purpose of the block in StageBlobs and StageGlobus. I'd recommend getting rid of the block support in these.

Reply:
@justinlittman I thought the same at first but then noticed that the AccessionWF would only be kicked off if the Staging succeeded?

Reply:
Ok, I've removed the block from both StageGlobus and StageBlobs and updated the tests. It does simplify the way the jobs read.